shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/TripDetailsStopList.kt

```diff
@@ -418,18 +446,19 @@ constructor(val tripId: String, val stops: List<Entry>, val startTerminalEntry:
 
         private fun getDisruption(
             entry: WorkingEntry,
+            fallbackTime: Instant?,
             alertsData: AlertsStreamDataResponse?,
             route: Route?,
             tripId: String,
             directionId: Int
         ): UpcomingFormat.Disruption? {
-            val entryTime = (entry.prediction ?: entry.schedule)?.stopTime
+            val entryTime = (entry.prediction ?: entry.schedule)?.stopTime ?: fallbackTime
             val entryRouteType = route?.type
 
-            if (entryTime == null) return null
             val alert =
                 alertsData?.alerts?.values?.find { alert ->
-                    alert.isActive(entryTime) &&
+                    (entryTime?.let { alert.isActive(it) }
+                        ?: true) &&
```

Comment:
This change allows us to actually show, for example, a stop closure at Bowdoin even though Bowdoin is arrival-only so there’s no trip time.